File: C#/8_Binary_Search.cs
// Solution 1: Binary Search Implementation
public class Solution {
    public int Search(int[] nums, int target) {
        int lpos = 0, rpos = nums.Length - 1, mid = 0;
        while (lpos <= rpos) {
            mid = (lpos + rpos) / 2;
            if (nums[mid] == target) 
                return mid;
            if (nums[mid] < target)
                lpos = mid + 1;
            else
                rpos = mid - 1;
        }
        return -1;
    }
}

// Solution 2: Using Array.BinarySearch
public class Solution {
    public int Search(int[] nums, int target) {
        int index = Array.BinarySearch(nums, target);
        if (index < 0)
            return -1;
        else
            return index;
    }
}

File: Python3/8_Binary_Search.py
from typing import List

class Solution:
    def search(self, nums: List[int], target: int) -> int:
        """
        Binary search algorithm to find the target in a sorted list.

        Args:
            nums: A list of integers representing the sorted array.
            target: The integer value to search for.

        Returns:
            The index of the target in the list if found, otherwise -1.
        """
        lpos: int = 0
        rpos: int = len(nums) - 1
        while lpos <= rpos:
            mid: int = (lpos + rpos) // 2
            if nums[mid] == target: 
                return mid
            if nums[mid] < target:
                lpos = mid + 1
            else:
                rpos = mid - 1
        return -1
